llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -1448,16 +1606,53 @@ void AddressSanitizer::instrumentSyclAllocateLocalMemory(CallInst *CI) {
       IRB.CreateCall(CI->getCalledFunction(),
                      {SizeWithRedZone, ConstantInt::get(IntptrTy, Align)});
 
-  /// __asan_set_shadow_local_memory(uptr beg, size_t size, size_t
-  /// size_with_redzone)
+  // __asan_set_shadow_static_local(
+  //   uptr beg,
+  //   size_t size,
+  //   size_t size_with_redzone,
+  //   uptr launch_info
+  // )
   IRB.CreateCall(
-      AsanSetShadowDeviceLocalFunc,
+      AsanSetShadowStaticLocalFunc,
       {IRB.CreatePointerCast(NewCI, IntptrTy), Size, SizeWithRedZone});
 
   CI->replaceAllUsesWith(NewCI);
   CI->eraseFromParent();
 }
 
+// Instument dynamic local memory
+void AddressSanitizer::instrumentSyclDynamicLocalMemory(Function &F) {
+  InstrumentationIRBuilder IRB(F.getEntryBlock().getFirstNonPHI());
```

Comment:
[`BasicBlock::getFirstInsertionPt`](https://www.llvm.org/docs/doxygen/classllvm_1_1BasicBlock.html#a161fd4e9fa5367f64c2a4c9e921c3ad3) is probably a better option here

Reply:
Its return type is not suitable for InstrumentationIRBuilder.